packages/node_modules/pouchdb-utils/src/toPromise.js

```diff
@@ -2,27 +2,15 @@ import Promise from 'pouchdb-promise';
 import getArguments from 'argsarray';
 import clone from './clone';
 import once from './once';
-import nextTick from './nextTick';
 
 function toPromise(func) {
   //create the function we will be returning
   return getArguments(function (args) {
     // Clone arguments
     args = clone(args);
     var self = this;
-    var tempCB =
-      (typeof args[args.length - 1] === 'function') ? args.pop() : false;
     // if the last argument is a function, assume its a callback
-    var usedCB;
-    if (tempCB) {
-      // if it was a callback, create a new callback which calls it,
-      // but do so async so we don't trap any errors
-      usedCB = function (err, resp) {
-        nextTick(function () {
-          tempCB(err, resp);
-        });
-      };
-    }
```

Comment:
This was the part of the code where we were most frequently (and most unnecessarily) calling `setTimeout()`. Note that because `tempCB` is called within a `then()`/`catch()` function, there is no reason for the extra indirection. We're already inside a microtask, we're already avoiding releasing Zalgo, so just calling the function directly is fine. Our tests still fully pass even with this change.